### PR TITLE
sds: simplify implementation to reduce dependencies

### PR DESCRIPTION
**Pull request by kyessenov**
Issue: #50134 

This brings down the binary size for pilot-agent to under 30MB.

CC @dpasiukevich

### CODE REVIEW

**Comment by kyessenov** [comment]
I don't quite understand where `/debug/syncz` is implemented in the agent - I thought I dropped discovery server debug handlers.

**Comment by howardjohn** [comment]
The syncz is https://github.com/istio/istio/blob/903ba1a06b4bacd8448cd7da10630b065cb4b8d1/pkg/istio-agent/xds_proxy.go#L787. its not implementing syncz itself, its a proxy server to Istiod

**Comment by kyessenov** [comment]
@howardjohn OK, that will pull all the Envoy protos used in listener, cluster, route, and extensions -- something we don't want to do just for debug endpoint. I removed that test since it doesn't seem to be critical -- you can decode Protos out of band.

**Comment by kyessenov** [comment]
@syw14 I think this change is fine because `istioctl x proxy-status --xds-via-agents` is still passing. On the wire, agent is sending the content base64 encoded Any, which `istioctl` is capable of decoding.

**Comment by kyessenov** [comment]
@igsong confirmed this change doesn't break `gcloud debug` command. Thank you @igsong .

File: pilot/pkg/xds/delta_test.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ import (
 	"istio.io/istio/pkg/test/util/assert"
 	"istio.io/istio/pkg/util/sets"
 	"istio.io/istio/pkg/workloadapi"
+	xdsserver "istio.io/istio/pkg/xds"
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/delta_test.go:36]:
```suggestion
	"istio.io/istio/pkg/xds"
```

File: pilot/pkg/xds/sds_test.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ import (
 	"istio.io/istio/pkg/spiffe"
 	"istio.io/istio/pkg/test/env"
 	"istio.io/istio/pkg/util/sets"
+	xdsserver "istio.io/istio/pkg/xds"
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/sds_test.go:44]: ditto

File: security/pkg/nodeagent/sds/sdsservice.go [hunk 3]
```diff
@@ -168,8 +121,10 @@ func newSDSService(st security.SecretManager, options *security.Options, pkpConf
 	return ret
 }
 
-func (s *sdsservice) generate(resourceNames []string) (model.Resources, error) {
-	resources := model.Resources{}
+var version uberatomic.Uint64
+
+func (s *sdsservice) generate(resourceNames []string) (*discovery.DiscoveryResponse, error) {
```
Comment (hzxuzhonghu) [inline security/pkg/nodeagent/sds/sdsservice.go:126]: feel a little duplicate if we write a bunch of methods again

File: pilot/pkg/xds/delta_test.go (hunk 1 above)
Comment (kyessenov) [reply pilot/pkg/xds/delta_test.go:36]: Package name conflict.

File: security/pkg/nodeagent/sds/sdsservice.go (hunk 3 above)
Comment (kyessenov) [reply security/pkg/nodeagent/sds/sdsservice.go:126]: You mean the version? Yes, but it's just two lines. It would be more code if we share.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**